packages/zosfiles/__tests__/__system__/cli/mount/fs/cli.files.mount.fs.system.test.ts

```diff
@@ -0,0 +1,169 @@
+import { getUniqueDatasetName, runCliScript } from "../../../../../../../__tests__/__src__/TestUtils";
+import { TestEnvironment } from "../../../../../../../__tests__/__src__/environment/TestEnvironment";
+import { ITestEnvironment } from "../../../../../../../__tests__/__src__/environment/doc/response/ITestEnvironment";
+import { ITestPropertiesSchema } from "../../../../../../../__tests__/__src__/properties/ITestPropertiesSchema";
+import { Delete } from "../../../../../src/api/methods/delete";
+
+const ZOWE_OPT_BASE_PATH = "ZOWE_OPT_BASE_PATH";
+
+let REAL_SESSION: Session;
+// Test Environment populated in the beforeAll();
+let TEST_ENVIRONMENT: ITestEnvironment;
+let TEST_ENVIRONMENT_NO_PROF: ITestEnvironment;
+let defaultSystem: ITestPropertiesSchema;
+let volume: string;
+
+describe("Mount and unmount file system", () => {
+    let fsname: string;
+    let mountPoint: string;
+
+    // Create the unique test environment
+    beforeAll(async () => {
+        TEST_ENVIRONMENT = await TestEnvironment.setUp({
+            tempProfileTypes: ["zosmf"],
+            testName: "zos_mount_unmount_file_system"
+        });
+
+        defaultSystem = TEST_ENVIRONMENT.systemTestProperties;
+        volume = defaultSystem.datasets.vol;
+
+        REAL_SESSION = TestEnvironment.createZosmfSession(TEST_ENVIRONMENT);
+        fsname = getUniqueDatasetName(defaultSystem.zosmf.user);
+
+        const dirname = getUniqueDatasetName(defaultSystem.zosmf.user).split(".")[1];
+        mountPoint = "//tmp/" + dirname;
+        const sshCommand = "mkdir " + mountPoint;
+
+        const response = runCliScript(__dirname + "/__scripts__/command/command_setup.sh",
+            TEST_ENVIRONMENT, [sshCommand, fsname,
+                defaultSystem.ssh.host,
+                defaultSystem.ssh.port,
+                defaultSystem.ssh.user,
+                defaultSystem.ssh.password]);
+    });
+
+    afterAll(async () => {
+        const sshCommand = "rmdir " + mountPoint;
+        const response = runCliScript(__dirname + "/__scripts__/command/command_teardown.sh",
+            TEST_ENVIRONMENT, [sshCommand, fsname,
+                defaultSystem.ssh.host,
+                defaultSystem.ssh.port,
+                defaultSystem.ssh.user,
+                defaultSystem.ssh.password]);
+
+        await TestEnvironment.cleanUp(TEST_ENVIRONMENT);
+    });
+
+    describe("without profiles", () => {
+        let defaultSys: ITestPropertiesSchema;
+
+        // Create the unique test environment
+        beforeAll(async () => {
+            TEST_ENVIRONMENT_NO_PROF = await TestEnvironment.setUp({
+                testName: "zos_files_mount_unmount_file_system_without_profile"
+            });
+
+            defaultSys = TEST_ENVIRONMENT_NO_PROF.systemTestProperties;
+        });
+
+        afterEach(async () => {
+            await TestEnvironment.cleanUp(TEST_ENVIRONMENT_NO_PROF);
+        });
+
+        it("should mount a file system", () => {
```

Comment:
Same comment here as the other test. They should be self contained and not depend on one another.